### ISSUE TITLE
Parser allows optional chain in a type-location - `class ... implements` and `interface ... extends`

### ISSUE THREAD

**Issue by bradzacher** [microsoft/TypeScript#41420]
**TypeScript Version:**  4.2.0-dev.20201105

**Search Terms:**
class implements optional chain
interface extends optional chain

**Code**

```ts
namespace FG {
    export class A {
        prop: string = '';
    }
}

class AA extends FG.A {}
type B = FG.A;

// @ts-expect-error -- Class 'C' incorrectly implements class 'A'. ... Property 'prop' is missing in type 'C' but required in type 'A'.
class C implements FG?.A {}

interface D extends FG?.A {}
// @ts-expect-error -- Property 'prop' is missing in type '{}' but required in type 'D'.
const test: D = {};

// expected parser error -- ';' expected.
type E = FG?.A;
```

**Expected behavior:**
```ts
class C implements FG?.A {} // expected a parser error
interface D extends FG?.A {} // expected a parser error
```

**Actual behavior:**

No parser errors - syntax is allowed

**Playground Link:** https://www.typescriptlang.org/play?ts=4.2.0-dev.20201105#code/HYQwtgpgzgDiDGEAEAxA4kg3gKCXpEAHjAPYBOALkvADYhRRICCWu+7MZJMAXElBTIBLYAHMkAXiQByaQG42eAL7YV2WvUZMWRChGAATRugB0LTCooBPGMgBCk1GjMLsAejdIAAhSgBaIlt4CgCyLjIkPz8kAGE6BhkY6SQReHIyCGCaKxSwGBoISGBfanjGaSZpEyQTWqQABS5bShzpTm5koUYwLqgRcREka1tE5IAjAFcqDIBHCaEMgxTgIZtkCqr1Mtjc-ML9EvQAfjMsNRE9MgAzBGQAEQJCPUNjNBPzFQ9vXwDiTJCIGFyJFoo1uIDrDJ2jBOt1ev1lqsRtILOMpkhZvNFojhus7ps0sABENoBQ+A8pBZXF9Av8IEs4GQoICCECIlEZHJkrTgvSTNhcUgAKKOY4ubBAA

**Related Issues:**
